app/src/main/java/com/amazonaws/kinesisvideo/demoapp/fragment/StreamWebRtcConfigurationFragment.java

```diff
@@ -181,6 +186,19 @@ private void setRegionFromCognito() {
         }
     }
 
+    private String generateDualStackEndpoint(String region) {
+        if (region == null || region.isEmpty()) {
+            Log.w(TAG, "AWS region is null or empty, will use legacy control-plane endpoint.");
```

Comment:
I'm not sure `region == null or empty` case is handled in the current app (aws clients normally have region as required). Could you double-check?

Reply:
That's right, create client fails cleanly with a "No region provided" alert displayed in the app, but might make sense to keep this check here if this will be treated as a stand alone util function.